hw/arc/hsdk_smp.c

```diff
@@ -0,0 +1,124 @@
+/*
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "qemu/osdep.h"
+#include "qemu/units.h"
+#include "qapi/error.h"
+#include "boot.h"
+#include "hw/boards.h"
+#include "hw/char/serial.h"
+#include "exec/address-spaces.h"
+#include "sysemu/reset.h"
+#include "sysemu/sysemu.h"
+#include "hw/arc/cpudevs.h"
+#include "hw/sysbus.h"
+#include "arconnect.h"
+
+#define HSDK_RAM_BASE      0x80000000
+#define HSDK_IO_BASE       0xf0000000
+#define HSDK_IO_SIZE       0x10000000
+
+/* UART */
+#define HSDK_UART_NUMBER   1
+#define HSDK_UART_OFFSET   0x5000
+#define HSDK_UART_BASE     (HSDK_IO_BASE + HSDK_UART_OFFSET)
+#define HSDK_UART_IRQ      (EXT_IRQ + 6)
+#define HSDK_UART_SIZE     0x2000
+
+/* Test Device */
+#define HSDK_TEST_OFFSET   0xa00000
+#define HSDK_TEST_BASE     (HSDK_IO_BASE + HSDK_TEST_OFFSET)
+
+static void hsdk_init(MachineState *machine)
+{
+    static struct arc_boot_info boot_info;
+    unsigned int smp_cpus = machine->smp.cpus;
```

Comment:
Do we really want to mimic a real board, but have that variability of cores used? Agian, maybe at least set default to 4 and not allow to specify more than 4?